pkg/source/openstack/client.go

```diff
@@ -528,11 +547,24 @@ func (c *Client) checkOrGetUUID(input string) (string, error) {
 		return "", fmt.Errorf("error extracting servers in checkorgetuuid:%v", err)
 	}
 
-	if len(allServers) > 1 {
-		return "", fmt.Errorf(NotUniqueName)
+	if len(allServers) == 0 {
+		return allServers[0].ID, nil
+	}
+
+	// api could return multiple servers matching the pattern of name
+	// eg server names test and testvm will match name search "test"
+	// in which case we need to filter on actual name
+	var filteredServers []servers.Server
+	for _, v := range allServers {
+		if v.Name == input {
+			filteredServers = append(filteredServers, v)
+		}
 	}
 
-	return allServers[0].ID, nil
+	if len(filteredServers) > 1 {
+		return "", fmt.Errorf(NotUniqueName)
+	}
+	return filteredServers[0].ID, nil
```

Comment:
Seem it'll also have index out of range panic when `allServers` is empty.

Reply:
this should not happen, the method looks for input name from import crd which can be vm uuid or name. As comment indicates searching for name `test` will return any vm with prefix `test` in it, so there is definitely a test instance in there, and the check is to ignore the additional vm's where the input is just a prefix.

so we should should always have atleast 1 vm in the result set.